alarm/PolicyManager2.js

```diff
@@ -1020,13 +1023,19 @@ class PolicyManager2 {
         break;
 
       case "category":
-        await Block.setupRules(scope && pid, target, "hash:ip", whitelist);
-        await Block.addMacToSet(scope, Block.getMacSet(pid));
-
-        if (!scope && !whitelist && target === 'default_c') try {
-          await categoryUpdater.iptablesRedirectCategory(target)
-        } catch(err) {
-          log.error("Failed to redirect default_c traffic", err)
+        if(policy.dnsmasq_entry){
+          await domainBlock.blockCategory(target,{
```

Comment:
Need to dynamically update domain lists in dnsmasq config as well. Looks like this function only fetches the domain list when the policy is created.

Reply:
Yes, I am doing that.